### PR TITLE
Accordion indicator color

### PR DESCRIPTION
**Pull request by lukelooker**
### :sparkles: Changes

- Add support for AccordionDisclosure `color` prop
- Tie AccordionDisclosure indicator color to AccordionDisclosure color
- Rename AccordionDisclosureLayoutGrid to remove `Grid` since it's no longer an actual grid layout (silly IE11)

### :white_check_mark: Requirements

- [ ] Includes test coverage for all changes
- [x] Build and tests are passing
- [x] Update documentation
- [x] Updated CHANGELOG
- [ ] Checked for i18n impacts
- [ ] Checked for a11y impacts
- [x] PR is ideally < ~400LOC

### CODE REVIEW

**Review by ElliotTheHuman — DISMISSED** [review]
LGTM!

**Review by ElliotTheHuman — APPROVED** [review]
LGTM (again)!

**Comment by lukelooker** [comment]
@ElliotTheHuman Took another look at this and came up with a much simpler solution - indicator now simply follows the color of the AccordionDisclosure itself. 

I added support for `color` prop on `AccordionDisclosure` as well to make it easy to use & test.

File: storybook/src/Accordion.stories.tsx [hunk 1]
```diff
@@ -84,7 +85,7 @@ export const DefaultOpen = () => (
 
 export const CustomizedIndicator = () => (
   <Accordion
-    indicatorSize="xxlarge"
+    // indicatorSize="xxlarge"
```
Comment (ElliotTheHuman) [inline storybook/src/Accordion.stories.tsx:88]: Nit: Think we can just delete this right? Doesn't have to be in this PR.

Comment (lukelooker) [reply storybook/src/Accordion.stories.tsx]: Whoops... that was actually a mistake. Reverted change altogether.

**Review by ElliotTheHuman — DISMISSED** [review]
LGTM

**Review by ElliotTheHuman — APPROVED** [review]
MTGL

**Pull request merged**